ledger/participant-integration-api/src/test/suite/scala/platform/store/packagemeta/PackageMetadataViewSpec.scala

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.packagemeta
+
+import com.daml.bazeltools.BazelRunfiles.rlocation
+import com.daml.daml_lf_dev.DamlLf
+import com.daml.ledger.test.SemanticTestDar
+import com.daml.lf.archive.DarParser
+import com.daml.lf.data.Ref
+import com.daml.platform.store.packagemeta.PackageMetadataView.MetadataDefinitions
+import com.daml.scalatest.FlatSpecCheckLaws
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import scalaz.scalacheck.ScalazProperties
+import com.daml.lf.value.test.ValueGenerators.idGen
+import com.daml.platform.store.packagemeta.PackageMetadataViewSpec._
+import org.scalatest.EitherValues
+
+import java.io.{File, FileInputStream}
+import java.util.zip.ZipInputStream
+import scala.util.Using
+
+class PackageMetadataViewSpec
+    extends AnyFlatSpec
+    with Matchers
+    with FlatSpecCheckLaws
+    with EitherValues {
+
+  behavior of "MetadataDefinitions Monoid"
+  checkLaws(ScalazProperties.monoid.laws[MetadataDefinitions])
+
+  behavior of "MetadataDefinitions from"
```

Comment:
how about not testing this at all?
the tested logic is whether we call the function right, and whether we put results in the right place. I would not even add a unit test for that, but would rather add a unit test (plus injecting the functionality), than testing this way with the LF utility.
It seems to me this is rather a subsystem/component test, and as such conveys little value, since these code-paths will be tested anyway by the conformance test.
(and ofc the LF utility has unit tests of it's own)

Also: not testing this would also make the daml file and the rumble in the BUILD.bazel also go away.

Reply:
yeah I was not sure about this either, but I had a lot of fun implementing it as it was the main complexity of this PR ;)